core/types/src/key_request.rs

```diff
@@ -72,10 +72,82 @@ bitflags! {
 #[derive(Debug, Clone, Hash, PartialEq, Eq)]
 #[repr(transparent)]
 pub struct KeyRequest(sgx_key_request_t);
+
 new_type_accessors_impls! {
     KeyRequest, sgx_key_request_t;
 }
 
+impl From<&[u8; 512]> for KeyRequest {
+    // Using `512` here instead of `sgx_key_request_t` to avoid leaking the
+    // abstraction to users
+    fn from(bytes: &[u8; 512]) -> Self {
+        // Unfortunately the `sgx_key_request_t` is not packed, as such we're
+        // relying on compiler convention as an assurance that the expected size
+        // is 512 bytes.
+        static_assertions::assert_eq_size!([u8; 512], sgx_key_request_t);
```

Comment:
After working on #126 and #127, I'm thinking this PR/part of the implementation should change.
I'm thinking `SealedData` is generic over an `AsRef<[u8]>` and that's more or less it.  It's an opaque structure that can be passed around to the C interface, and nothing else.  TBH I'm not sure why it isn't an opaque structure in the C api.

I stayed away from the TryFrom since 512 doesn't account for the payload, but I think I better understand the payload and mac(aad) offset enough to say that we should be able to implement a tryfrom

Reply:
implemented tryfrom